sys/include/clif.h

```diff
@@ -0,0 +1,218 @@
+/*
+ * Copyright (C) 2019 HAW Hamburg
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    sys_clif CoRE Link Format
+ * @ingroup     sys_serialization
+ * @brief       Simple encoding and decoding of CoRE Link Format strings
+ *
+ * @{
+ *
+ * @file
+ * @brief       CoRE Link Format encoding and decoding library public
+ *              definitions
+ *
+ * @author      Leandro Lanzieri <leandro.lanzieri@haw-hamburg.de>
+ */
+
+#ifndef CLIF_H
+#define CLIF_H
+
+#include <sys/types.h>
+
+#include "clif_internal.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief Return types for the @ref sys_clif API
+ */
+enum {
+    CLIF_OK          = 0,    /**< success */
+    CLIF_NO_SPACE    = -1,   /**< not enough space in the buffer */
+    CLIF_NOT_FOUND   = -2    /**< could not find a component in a buffer */
+};
+
+/**
+ * @brief Types of link format parameters
+ */
+typedef enum {
+    CLIF_PARAM_ANCHOR        = 0,  /**< anchor */
+    CLIF_PARAM_REL           = 1,  /**< rel */
+    CLIF_PARAM_LANG          = 2,  /**< hreflang */
+    CLIF_PARAM_MEDIA         = 3,  /**< media */
+    CLIF_PARAM_TITLE         = 4,  /**< title */
+    CLIF_PARAM_TITLE_EXT     = 5,  /**< title* */
+    CLIF_PARAM_TYPE          = 6,  /**< type */
+    CLIF_PARAM_RT            = 7,  /**< rt */
+    CLIF_PARAM_IF            = 8,  /**< if */
+    CLIF_PARAM_SZ            = 9,  /**< sz */
+    CLIF_PARAM_CT            = 10, /**< ct */
+    CLIF_PARAM_EXT           = 11  /**< extensions */
+} clif_param_type_t;
+
+/**
+ * @brief Link format parameter descriptor
+ */
+typedef struct {
+    clif_param_type_t type;        /**< type of parameter */
```

Comment:
Is it worthwhile to include this here and generate it with each call to `clif_get_param()`? Maybe create a separate `unsigned clif_get_param_type(const char *key)` and let the user decide when to use it.

In this case the `char *ext` attribute becomes `char *key`.

Reply:
Sounds reasonable, I will change this.